src/main/java/org/opentripplanner/common/model/Result.java

```diff
@@ -0,0 +1,139 @@
+package org.opentripplanner.common.model;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+
+/**
+ * A type for containing either a success or a failure type as the result of a computation.
+ * <p>
+ * It's very similar to the Either or Validation type found in functional programming languages.
+ */
+public abstract class Result<L, R> {
+
+  private Result() {}
+
+  public static <L, R> Result<L, R> failure(@Nonnull L failure) {
```

Comment:
Aren't the types usually reversed, so that the first one is the expected and the second one the error type?

Reply:
I took it from Scala's Validation: https://github.com/scalaz/scalaz/blob/series/7.1.x/core/src/main/scala/scalaz/Validation.scala

There the first type is the error and the second is the result.

I'm open to changing it though.

Reply:
I want to be more like Rust and less like Scala, so I swapped the order around.